src/library/dao/trackdao.cpp

```diff
@@ -1313,9 +1308,7 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
         TrackPopulatorFn populator = columns[i].populator;
         if (populator != nullptr) {
             // If any populator says the track should be dirty then we dirty it.
-            if ((*populator)(queryRecord, i, pTrack)) {
-                shouldDirty = true;
-            }
+            shouldDirty = (*populator)(queryRecord, i, pTrack);
```

Comment:
Why did you change this unrelated code and introduce a bug? It no longer works as intended.

Reply:
Ah, OK, I see. I will revert this. Thank you.

Reply:
Please note that I discovered this flaw by chance and didn't do a review of the remaining code.